src/mam/mod_mam_cassandra_prefs.erl

```diff
@@ -100,24 +102,26 @@ get_behaviour(DefaultBehaviour, HostType, _UserID, LocJID, RemJID) ->
                     {JID1, B1} < {JID2, B2}
                 end, Rows),
             #{behaviour := Behaviour} = lists:last(SortedRows),
-            decode_behaviour(Behaviour)
+            {ok, decode_behaviour(Behaviour)}
     end.
 
 
--spec set_prefs(Result :: any(), HostType :: host_type(),
-                ArchiveID :: mod_mam:archive_id(), ArchiveJID :: jid:jid(),
-                DefaultMode :: mod_mam:archive_behaviour(),
-                AlwaysJIDs :: [jid:literal_jid()],
-                NeverJIDs :: [jid:literal_jid()]) -> any().
```

Comment:
I'd be good if all these types were encoded and not lost. Perhaps an auxiliary type to declare in the same place the callback is declared, so that handlers can just use it.